src/sql/workbench/contrib/notebook/browser/notebookEditor.component.ts

```diff
@@ -0,0 +1,197 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { Component, Inject, ChangeDetectorRef, forwardRef } from '@angular/core';
+import { NotebookModel } from 'sql/workbench/services/notebook/browser/models/notebookModel';
+import * as notebookUtils from 'sql/workbench/services/notebook/browser/models/notebookUtils';
+import { AngularDisposable } from 'sql/base/browser/lifecycle';
+import { IBootstrapParams } from 'sql/workbench/services/bootstrap/common/bootstrapParams';
+import { INotebookParams, INotebookService, INotebookManager, DEFAULT_NOTEBOOK_PROVIDER, SQL_NOTEBOOK_PROVIDER } from 'sql/workbench/services/notebook/browser/notebookService';
+import { IConnectionManagementService } from 'sql/platform/connection/common/connectionManagement';
+import { CellMagicMapper } from 'sql/workbench/contrib/notebook/browser/models/cellMagicMapper';
+import { INotificationService } from 'vs/platform/notification/common/notification';
+import { ICapabilitiesService } from 'sql/platform/capabilities/common/capabilitiesService';
+import { IAdsTelemetryService } from 'sql/platform/telemetry/common/telemetry';
+import { ILogService } from 'vs/platform/log/common/log';
+import { IModelFactory, ViewMode, NotebookContentChange } from 'sql/workbench/services/notebook/browser/models/modelInterfaces';
+import { IConnectionProfile } from 'sql/platform/connection/common/interfaces';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { ModelFactory } from 'sql/workbench/services/notebook/browser/models/modelFactory';
+import { onUnexpectedError } from 'vs/base/common/errors';
+import { IContextKeyService, RawContextKey } from 'vs/platform/contextkey/common/contextkey';
+import { IAction } from 'vs/base/common/actions';
+import { IMenuService, MenuId } from 'vs/platform/actions/common/actions';
+import { fillInActions } from 'vs/platform/actions/browser/menuEntryActionViewItem';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { NOTEBOOK_VIEWS_ENABLED_PROPERTY } from 'sql/workbench/contrib/notebook/browser/notebookViews/notebookViews.contribution';
+
+export const NOTEBOOKEDITOR_SELECTOR: string = 'notebookeditor-component';
+
+@Component({
+	selector: NOTEBOOKEDITOR_SELECTOR,
+	templateUrl: decodeURI(require.toUrl('./notebookEditor.component.html'))
+})
+export class NotebookEditorComponent extends AngularDisposable {
+	private readonly defaultViewMode = ViewMode.Notebook;
+	private profile: IConnectionProfile;
+	private notebookManagers: INotebookManager[] = [];
+
+	private _model: NotebookModel;
+	public viewMode: ViewMode;
```

Comment:
This should be readonly I'd think. 

And also I would probably just hook it up to be a property getter that fetches the value from the model - right now they could be different values which would be odd to say the least.

Reply:
Makes sense. I made it a getter that fetches a value from the model.